File: ProsperBorrowerSDKSample/ProsperBorrowerSDKSample/OfferViewController.m
//
//  OfferViewController.m
//  ProsperBorrowerSDKSample
//
//  Copyright © 2015 Prosper Funding, LLC, All rights reserved.
//

#import "OfferViewController.h"

@interface OfferViewController ()
@property (weak, nonatomic) IBOutlet UIView *loadingView;
@end

@implementation OfferViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UITapGestureRecognizer *singleTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleTap:)];
    [self.view addGestureRecognizer:singleTap];
    
    self.navigationController.navigationBar.hidden = YES;
}

- (void)handleTap:(UITapGestureRecognizer*)gesture {
    
    PMIBorrowerViewController *prosperViewController = [[PMIBorrowerViewController alloc] initWithOffer:self.loanOffer delegate:self];
    [self presentViewController:prosperViewController animated:YES completion:nil];
}


#pragma mark - PMIBorrowerDelegate methods

- (void)borrowerViewController:(PMIBorrowerViewController *)borrowerViewController loanProcessedStatus:(BorrowerLoanStatus)status {
    NSString *loanStatus = @"";
    if(status == PMIBorrowerLoanSuccess) {
        loanStatus = @"Loan is successfully processed.";
    } else if(status == PMIBorrowerLoanCancelled) {
        loanStatus = @"Loan Process got cancelled";
    } else if(status == PMIBorrowerTimedOut) {
        loanStatus = @"Loan Process got timed out due to inactive user session";
        
    }
    
    UIAlertController * alertController =   [UIAlertController
                                             alertControllerWithTitle:@"Alert"
                                             message:loanStatus
                                             preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction *cancelAlertAction = [UIAlertAction actionWithTitle:@"Ok"
                                                                style:UIAlertActionStyleDefault
                                                              handler:nil];
    [alertController addAction:cancelAlertAction];
    
    [self presentViewController:alertController animated:YES completion:nil];
    
    [self.navigationController popToRootViewControllerAnimated:YES];
}

@end

File: ProsperSDKSwiftSample/ProsperSDKSwiftSample/OfferViewController.swift
//
//  OfferViewController.swift
//  ProsperSDKSwiftSample
//
//  Copyright © 2015 Prosper Funding, LLC, All rights reserved.
//

import UIKit
import ProsperBorrowerSDK

class OfferViewController: UIViewController, PMIBorrowerDelegate {

    var loanOffer:PMILoanOffer!

    @IBOutlet weak var loadingView: UIView!
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(OfferViewController.handleTap(_:)))
        self.view.addGestureRecognizer(tapGesture)
        self.navigationController?.navigationBar.isHidden = true

        // Do any additional setup after loading the view.
    }
    
    func handleTap(_ gestureRecognizer: UIGestureRecognizer) {
        // handling code
        let borrowerViewController =  PMIBorrowerViewController.init(offer: loanOffer, delegate: self)
        self.present(borrowerViewController!, animated: true, completion: nil)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func borrowerViewController(_ borrowerViewController:PMIBorrowerViewController, loanProcessedStatus loanStatus:BorrowerLoanStatus) {
        var status:String = ""
        if loanStatus == PMIBorrowerLoanSuccess {
            status = "Loan is successfully processed."
        } else if loanStatus == PMIBorrowerLoanCancelled {
            status = "Loan is cancelled."
        } else if loanStatus == PMIBorrowerTimedOut {
            status = "Loan Process got timed out due to inactive user session."
        }
        
        let alert = UIAlertController(title: "Alert",
                                      message: status,
                                      preferredStyle: UIAlertControllerStyle.alert)
        alert.addAction(UIAlertAction(title: "Ok", style: UIAlertActionStyle.default, handler: nil))
        
        self.present(alert, animated: true, completion: nil)
        
        
        _=navigationController?.popToRootViewController(animated: true)
    }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
